src/librustc/ty/context.rs

```diff
@@ -2441,7 +2441,7 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     /// type with the same signature. Detuples and so forth -- so
     /// e.g., if we have a sig with `Fn<(u32, i32)>` then you would get
     /// a `fn(u32, i32)`.
```

Comment:
Can you extend the doc-comment with note about the role of `unsafety`? Notably, the closure itself should not be able to do unsafe stuff.